### PR TITLE
Fix for static library export issue in blis package

### PR DESCRIPTION
**Pull request by amd-toolchain-support**

### CODE REVIEW

File: var/spack/repos/builtin/packages/blis/package.py [hunk 1]
```diff
@@ -111,8 +111,12 @@ def darwin_fix(self):
 
     @property
     def libs(self):
+        shared = True if '+shared' in self.spec else False
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/blis/package.py]: FWIW. Could simplify:

```suggestion
        shared = '+shared' in self.spec
```

File: var/spack/repos/builtin/packages/blis/package.py [hunk 2]
```diff
@@ -111,8 +111,12 @@ def darwin_fix(self):
...
         return find_libraries(
-            ["libblis", "libblis-mt"], root=self.prefix, recursive=True
+            ["libblis", "libblis-mt"],
+            root=self.prefix,
+            shared=shared,
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/blis/package.py]: Or you could simply do the following:

```suggestion
            shared='+shared' in self.spec,
```

**Comment by tldahlgren** [comment]
Is there any documentation (and associated URL) you could provide that explains the static library export issue?  If so, would be good to add it here.

File: var/spack/repos/builtin/packages/blis/package.py (hunk 1 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/blis/package.py]: Removed this line and used shared = '+shared' in self.spec within return find_libraries block.

File: var/spack/repos/builtin/packages/blis/package.py (hunk 2 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/blis/package.py]: Incorporated review comments

**Comment by amd-toolchain-support** [comment]
> Is there any documentation (and associated URL) you could provide that explains the static library export issue? If so, would be good to add it here.

while building VASP application with static amdscalapack libraries, using static amdblis and amdlibflame observed build failure.

To nail down the issue, we executed below command
spack install -v amdscalapack ~shared %aocc@3.0.0 ^amdblis@3.0 ~shared ^amdlibflame@3.0 ~shared

Post triaging we figured out that issue is with blis package (From which amdblis has inherited), during exporting static library.

**Review by tldahlgren — APPROVED** [review]
LGTM.

**Comment by tldahlgren** [comment]
Thank you for your contribution!

**Pull request merged**